ballerina/Module.md

```diff
@@ -1,17 +1,189 @@
 ## Overview
 
-[//]: # (TODO: Add overview mentioning the purpose of the module, supported REST API versions, and other high-level details.)
+[HubSpot ](https://www.hubspot.com/) is an AI-powered customer relationship management (CRM) platform. 
+
+The `ballerinax/hubspot.automation.actions` offers APIs to connect and interact with the [Automation Actions](https://developers.hubspot.com/docs/reference/api/automation/custom-workflow-actions)  endpoints, specifically based on the [API v4](https://developers.hubspot.com/docs/reference/api/automation/custom-workflow-actions)
 
 ## Setup guide
 
-[//]: # (TODO: Add detailed steps to obtain credentials and configure the module.)
+To use the Hubspot Automation action API connector in Ballerina, you must have a HubSpot developer account. If you don't have an account, create one using the following steps.
+
+### Step 1: Create a Hubspot Developer Account
+
+Visit the Hubspot portal (https://developers.hubspot.com/get-started) and create a Developer Account.
+
+### Step 2: Create a Hubspot Developer Test Account
+
+Visit (https://developers.hubspot.com/beta-docs/getting-started/account-types#developer-test-accounts)and create a hubspot developer test account
+
+### Step 3: Create a Hubspot Public App
+
+In your developer account, navigate to the "Apps" section.
+
+Click on "Create App" and provide the necessary details, including the app name and description.
+
+
+### Step 4: Initiate the OAuth Flow
+
+Move to the auth tab in the created app and set the permissions there.
+
+Under the OAuth tab you can find the following details,
+
+* `client_id`: Your app's Client ID.
+* `redirect_uri`: The URL users will be redirected to after granting access.
+* `scope`: A space-separated list of scopes your app is requesting.
+
+![Auth Tab example](docs/setup/resources/auth.png)
```

Comment:
These links won't work as the path to the image is incorrect. Please provide the relative path. Check other places as well.
```suggestion
![Auth Tab example](../docs/setup/resources/auth.png)
```